Move iris to use jinja templates

Jaeyoung-Lim opened the pull request:
**Problem Description**
`iris` was the only model being generated with xacro macros, and xacro macros are slowly becoming out of date by not supporting more recent sdf syntaxes.

**Solution**
This PR moves the `iris` model generation from xacro macros to jinja templates by doing the following.
- Deprecate xacro genration commands from cmake and rely on jinja templates
- Moves iris mesh files from `rotors_description` to `iris`

We keep the xacro macros for now, since the multivehicle scripts still rely on them. `rotors_description` will be deprecated once the multivehicle scripts are based on jinja templating.

Jaeyoung-Lim committed:
mark iris.sdf as iris-gen.sdf

Since generated sdf files are now marked as <model>-gen.sdf, `iris.sdf` should be also marked as `iris-gen.sdf`
Move iris to jinja templates

This commit deprecates the xacro based sdf generation for the iris model

Jaeyoung-Lim committed:
Move meshfiles out of rotors_description

This commit moves the meshfiles out of rotors_description so that we can deprecate rotors_description once the mutlivehicle scripts are based on jinja templates

Jaeyoung-Lim committed:
Update mavsdk version

Wasn't able to build mavsdk  tests due to the incorrect mavsdk version

TSC21 commented on models/iris/iris.sdf.jinja:
```diff
@@ -0,0 +1,619 @@
+<sdf version='1.7'>
```
We should add a note stating that the gen file was generated from this. Similar to https://github.com/PX4/sitl_gazebo/blob/master/models/matrice_100/matrice_100.sdf.jinja#L95

Jaeyoung-Lim replied:
Thanks for pointing that out! added on last commit

TSC21 commented on models/iris/iris.sdf.jinja:
```diff
@@ -0,0 +1,620 @@
+    </plugin>
+    <model name='gps0'>
+      <link name='link'>
+        <pose>0 0 0 0 -0 0</pose>
```
can we delete the '-0' throughout the file?

TSC21 commented on models/iris/iris.sdf.jinja:
```diff
@@ -0,0 +1,620 @@
+          <zero_position_armed>100</zero_position_armed>
+          <joint_control_type>velocity</joint_control_type>
+        </channel>
+        <channel name='rotor5'>
```
why is this rotor needed?

TSC21 commented on models/iris/iris.sdf.jinja:
```diff
@@ -0,0 +1,620 @@
+          </joint_control_pid>
+          <joint_name>zephyr_delta_wing::propeller_joint</joint_name>
+        </channel>
+        <channel name='rotor6'>
```
and this?

Jaeyoung-Lim replied:
Since this is not related to this PR, and is causing SITL failures can we address it in a separate PR? This is blocking multivehicle scripting and mavsdk testing for jinja based models

TSC21 replied:
But it's not making much sense why this would cause failures. This isn't supposed to exist for this model.

Jaeyoung-Lim committed:
Address review comments

- Explicitly state that the sdf file is genretaed
- Remove -0
- Remove redundant channels
- Fix model configs
